core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -325,13 +331,265 @@ STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_multiply(
+///                     gen: bytes, scratch_buffer: bytearray) -> memoryview:
+///     '''
+///     Return a range proof for specified value (as a memoryview).
+///     '''
+STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_rangeproof_sign(
+    size_t n_args, const mp_obj_t *args) {
+  // TODO: not sure about the constants values here.
+  const uint64_t VALUE_MIN = 1;
+  const int EXPONENT = 0;
+  const int BITS = 32;
+
+  const secp256k1_context *ctx =
+      mod_trezorcrypto_get_secp256k1_context(args[0]);
+  const uint64_t value = _mp_obj_get_uint64(args[1]);
+
+  mp_buffer_info_t commit;
+  mp_get_buffer_raise(args[2], &commit, MP_BUFFER_READ);
+  if (commit.len != 33) {
+    mp_raise_ValueError("Invalid length of commitment");
+  }
+  secp256k1_pedersen_commitment commitment;
+  if (secp256k1_pedersen_commitment_parse(ctx, &commitment, commit.buf) != 1) {
+    mp_raise_ValueError("Invalid Pedersen commitment");
+  }
+
+  mp_buffer_info_t blind;
+  mp_get_buffer_raise(args[3], &blind, MP_BUFFER_READ);
+  if (blind.len != 32) {
+    mp_raise_ValueError("Invalid length of blinding factor");
+  }
+
+  mp_buffer_info_t nonce;
+  mp_get_buffer_raise(args[4], &nonce, MP_BUFFER_READ);
+  if (nonce.len != 32) {
+    mp_raise_ValueError("Invalid length of nonce");
+  }
+
+  mp_buffer_info_t message;
+  mp_get_buffer_raise(args[5], &message, MP_BUFFER_READ);
+
+  mp_buffer_info_t extra_commit;
+  mp_get_buffer_raise(args[6], &extra_commit, MP_BUFFER_READ);
+
+  mp_buffer_info_t gen;
+  mp_get_buffer_raise(args[7], &gen, MP_BUFFER_READ);
+  if (gen.len != 33) {
+    mp_raise_ValueError("Invalid length of generator");
+  }
+  secp256k1_generator generator;
+  if (!secp256k1_generator_parse(ctx, &generator, gen.buf)) {
+    mp_raise_ValueError("Generator parsing failed");
+  }
+
+  mp_buffer_info_t scratch_buffer;
+  mp_get_buffer_raise(args[8], &scratch_buffer, MP_BUFFER_WRITE);
+  if (scratch_buffer.len < RANGEPROOF_SIGN_BUFFER_SIZE) {
+    mp_raise_ValueError("Invalid length of output buffer");
+  }
+
+  size_t rangeproof_len = scratch_buffer.len;
+  if (!secp256k1_rangeproof_sign(
+          ctx, scratch_buffer.buf, &rangeproof_len, VALUE_MIN, &commitment,
+          blind.buf, nonce.buf, EXPONENT, BITS, value, message.buf, message.len,
+          extra_commit.buf, extra_commit.len, &generator)) {
+    mp_raise_ValueError("Rangeproof sign failed");
+  }
+  return mp_obj_new_memoryview('B', rangeproof_len, scratch_buffer.buf);
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(
+    mod_trezorcrypto_secp256k1_context_rangeproof_sign_obj, 9, 9,
+    mod_trezorcrypto_secp256k1_context_rangeproof_sign);
+
+/// def rangeproof_rewind(self, conf_value: bytes, conf_asset: bytes,
+///                       nonce: bytes, range_proof: bytes,
+///                       extra_commit: bytes, asset_message: bytearray) ->
+///                       (value: long, blind: bytes,
+///                        asset_message: memoryview):
```

Comment:
nit: the "asset" in `asset_message" is maybe too specific. This function is generic and messages could be other things.

Reply:
You're right, thanks!
Fixed at https://github.com/romanz/trezor-firmware/pull/1/commits/729918821371de7462160ddc5e18f55944e26871